modin/experimental/engines/omnisci_on_native/test/test_dataframe.py

```diff
@@ -873,6 +874,32 @@ def dict_agg_all_cols(df, **kwargs):
 
         run_and_compare(dict_agg_all_cols, data=self.data)
 
+    # modin-issue#3461
+    def test_groupby_pure_by(self):
+        data = [1, 1, 2, 2]
+        md_ser, pd_ser = pd.Series(data), pandas.Series(data)
+
+        set_execution_mode(md_ser, "lazy")
+        md_res = md_ser.groupby(md_ser).sum()
+        set_execution_mode(md_res, None)
+
+        pd_res = pd_ser.groupby(pd_ser).sum()
+        df_equals(md_res, pd_res)
```

Comment:
Is there a reason to not use `run_and_compare` here?

Reply:
no reason, used `run_and_compare` in the last commit.